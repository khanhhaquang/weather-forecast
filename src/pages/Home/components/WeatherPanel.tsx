import { useSelector } from 'react-redux';
import { getWeatherData } from '../redux/reducer';
import WeatherCard from './WeatherCard';

const WeatherPanel: React.FC = () => {
  const weatherArray = useSelector(getWeatherData);

  return (
    <>
      {weatherArray.slice(0, 5).map((data) => (
        <WeatherCard
          data-testid={`weather-card-${data.id}`}
          key={data.id}
          minTemp={data.min_temp}
          maxTemp={data.max_temp}
          stateImg={data.weather_state_abbr}
          stateName={data.weather_state_name}
          day={data.applicable_date}
        />
      ))}
    </>
  );
};

export default WeatherPanel;
