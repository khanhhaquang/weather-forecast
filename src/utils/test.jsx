/* eslint-disable import/no-unresolved */
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { render as rtlRender } from '@testing-library/react';
import rootReducer from 'pages/reducer';

function render(ui, { initialState, store = createStore(rootReducer, initialState) } = {}) {
  return {
    ...rtlRender(<Provider store={store}>{ui}</Provider>),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store,
  };
}

export * from '@testing-library/react';
export { render };
