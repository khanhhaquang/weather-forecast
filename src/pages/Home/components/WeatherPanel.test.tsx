import { render, screen } from 'utils/test';
import WeatherPanel from './WeatherPanel';

describe('Home/WeatherPanel', () => {
  const mockState = {
    home: {
      data: [
        {
          id: 1,
          min_temp: 10,
          max_temp: 20,
          applicable_date: '1999/1/1',
          weather_state_name: 'foo',
          weather_state_abbr: 'bar',
        },
      ],
    },
  };

  it('should render correctly', () => {
    const { asFragment } = render(<WeatherPanel />, {
      initialState: { ...mockState },
    });
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByTestId('weather-card-1')).toBeInTheDocument();
  });
});
