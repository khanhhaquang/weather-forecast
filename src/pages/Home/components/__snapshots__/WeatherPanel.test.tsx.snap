// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home/WeatherPanel should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnxRM gCZTqE"
    data-testid="weather-card-1"
  >
    <h2
      class="sc-gtsrHT ebtAAb"
    >
      Friday
    </h2>
    <div
      class="sc-dlnjwi QbBjM"
    >
      <img
        alt="state"
        height="24"
        src="https://meta-weather.now.sh/static/img/weather/png/64/bar.png"
        width="24"
      />
      <span>
        foo
      </span>
    </div>
    <span>
      Max: 20°C
    </span>
    <span>
      Min: 10°C
    </span>
  </div>
</DocumentFragment>
`;
